### PR TITLE
De-duplicate some functions

### PR DESCRIPTION
**Pull request by james-d-mitchell**
This PR contains some more refactoring of the GAP library code to remove duplicate functions. The functions touched in this PR have identical function bodies (up to whitespace).

### CODE REVIEW

File: lib/init.g [hunk 1]
```diff
@@ -406,35 +406,35 @@ CallAndInstallPostRestore( function()
   MakeImmutable( GAPInfo.TermEncoding );
 end );
 
-
-
-BindGlobal( "ShowKernelInformation", function()
-  local sysdate, linelen, indent, btop, vert, bbot, print_info,
-        config, str, gap;
+BindGlobal("PrintInfo",
```
Comment (fingolfin) [inline lib/init.g:409]: That's a pretty generic name, maybe not the best to use for an internal helper? Perhaps `_PrintInfo` or s/o?

File: lib/ffeconway.gi [hunk 2]
```diff
@@ -1184,33 +1184,6 @@ InstallMethod( LogFFE,
         [ IsFFE and IsCoeffsModConwayPolRep, IsFFE and IsInternalRep],
         DoDLog );
 
-#############################################################################
-##
-#M  Order -- multiplicative order of an FFE
-##
-
-
-InstallMethod( Order,
```
Comment (fingolfin) [inline lib/ffeconway.gi:1193]: Where is the code that is making this one redundant?


Also, can you please apply the same change to `hpcgap/lib/ffeconway.gi`?

File: lib/ring.gi [hunk 3]
```diff
@@ -921,18 +921,6 @@ InstallMethod( Units,
     if IsAssociative(R) then
       expo:=1;
 
-      # what power until you get idempotent (counting 1/0 as idempotent)
-      idempo:=function(elm)
```
Comment (fingolfin) [inline lib/ring.gi:925]: Oh wow, it is assigned to the same just a few lines earlier, great catch :-)

File: lib/grppc.gi [hunk 4]
```diff
@@ -1491,18 +1491,7 @@ local G,           # common parent
...
-                  for j in Npcgs do
-                    if DepthOfPcElement(pcgs,Comm(j,i))<dep then
-                      return false;
-                    fi;
-                  od;
-                od;
-                return true;
-              end;
-
+        cent:=PcClassFactorCentralityTest;
```
Comment (fingolfin) [inline lib/grppc.gi:1494]: Another great catch -- out of curiosity, how did you find this?

File: lib/algebra.gi [hunk 5]
```diff
@@ -2843,10 +2843,9 @@ InstallMethod( Basis,
 ##
 #M  Basis( <A> )  . . . . . .  basis from FLMLOR gens. for associative FLMLOR
 ##
-InstallMethod( Basis,
-    "for an associative FLMLOR",
-    [ IsFLMLOR and IsAssociative ],
-        function( A )
+
```
Comment (fingolfin) [inline lib/algebra.gi:2846]:
```suggestion
```

File: lib/algebra.gi [hunk 6]
```diff
@@ -2903,28 +2905,11 @@ InstallMethod( Basis,
 ##  this means that it is sufficient to multiply with generators from one
 ##  side.
 ##
+
```
Comment (fingolfin) [inline lib/algebra.gi:2908]:
```suggestion
```

File: lib/alglie.gi [hunk 7]
```diff
@@ -3776,6 +3776,45 @@ end );
 ##  For the algorithm we refer to C. Reutenauer, Free Lie Algebras, Clarendon
 ##  Press, Oxford, 1993.
 ##
+
+DeclareGlobalFunction( "IsLyndonT" );
+
+InstallGlobalFunction( IsLyndonT,
```
Comment (fingolfin) [inline lib/alglie.gi:3782]: I would prefer not to introduced new uses of `DeclareGlobalFunction` (which introduces a "placeholder" object) esp. for internal helpers. This should work, too:
```suggestion
DeclareGlobalName( "IsLyndonT" );

BindGlobal( "IsLyndonT",
```

**Review by fingolfin — COMMENTED** [review]
Thanks, very nice! A few quick questions

File: lib/grppc.gi (hunk 4 above)
Comment (james-d-mitchell) [reply lib/grppc.gi:1494]: The next version of gaplint warns about duplicate functions, I'm using that, there about 80 more duplicates in the library with at least 4 lines. Thanks for the feedback, I'll fix this up tomorrow @fingolfin

File: lib/ffeconway.gi (hunk 2 above)
Comment (james-d-mitchell) [reply lib/ffeconway.gi:1193]: The method is duplicated in `ffe.gi:779`. And yes, will do!

File: lib/memory.gi [hunk 8]
```diff
@@ -217,22 +217,7 @@ InstallMethod( \*, "objects with memory", true,
...
-      if IsMatrix(a) then
-        SetFilterObj(r,IsMatrix);
-      fi;
-      if IsMatrixObj(a) then
-        SetFilterObj(r,IsMatrixObj);
-      fi;
-    fi;
-    return r;
-  end);
+  [IsObjWithMemory],0, OneOp);
```
[the remark this answers is not in the document]
Comment (james-d-mitchell) [reply lib/memory.gi:220]: True, but is it likely to be critical anywhere? I can reverse this if desirable.

[the remark this answers is not in the document]
Comment (fingolfin) [reply lib/memory.gi:220]: Probably not critical, aye.

**Review by fingolfin — APPROVED** [review]

**Pull request merged**